### PR TITLE
fix for issue #1205, increase event count only for valid events, that…

### PR DESCRIPTION
**Pull request by dtacalau**
… is events for sockets which have not been deleted

Signed-off-by: Daniel Tacalau <dst4096@gmail.com>

### CODE REVIEW

File: tests/poll.rs [hunk 1]
```diff
@@ -201,6 +204,49 @@ fn registry_behind_arc() {
     handle2.join().unwrap();
 }
 
+#[test]
+fn poll_no_timeout_not_affected_by_deregister() {
```
Comment (Thomasdezeeuw) [inline tests/poll.rs]: Can you move this to the `regressions.rs` file, then the comment below won't be needed.

File: tests/poll.rs [hunk 2]
```diff
@@ -201,6 +204,49 @@ fn registry_behind_arc() {
...
+    // Test to verify issue 1205 does not reproduce again
+    init();
```
Comment (Thomasdezeeuw) [inline tests/poll.rs]: You can use `init_with_poll`, then you don't need create the `poll` and `events` below.

File: tests/poll.rs (hunk 1 above)
Comment (dtacalau) [reply tests/poll.rs]: done

File: tests/poll.rs (hunk 2 above)
Comment (dtacalau) [reply tests/poll.rs]: done

**Comment by Thomasdezeeuw** [comment]
Thanks.

**Pull request merged**